src/EventHint.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry;
+
+/**
+ * This class represents hints on how to process an event.
+ */
+final class EventHint
```

Comment:
While I absolutely love typed things, I don't think that in this case we should typehint the `$hint` argument, mainly because it should allow users to carry any extra data they want and in this case it's not possible without setting a property dinamically (class is `final` and the `fromArray` method does not accept unknown properties) that makes this object much more similar to a plain array

Reply:
Can we leave it typed but not throw an exception and just discard unknown args?
I think that might be a better solution.

Reply:
>Can we leave it typed but not throw an exception and just discard unknown args?

Of course we can, but I really can't find any good reason to have an object and then treat it like a plain array like in this case. With [array shapes](https://phpstan.org/writing-php-code/phpdoc-types#array-shapes) we can document the known keys that should be in the array so that static analyzers understand the minimal payload that the argument can carry, and for the rest it's up to the user what to pass and we shouldn't put any limit on it